File: frontend/src/models/ScheduleRequest.ts
import { Worker } from './Worker';
import { Shift } from './Shift';
import { Rule } from './Rule';

export class ScheduleRequest {
    workers: Worker[];
    shifts: Shift[];
    month: number;
    fullTimeHours: number;
    rules: Rule[];

    constructor(
        workers: Worker[] = [],
        shifts: Shift[] = [],
        month: number,
        fullTimeHours: number,
        rules: Rule[] = []
    ) {
        this.workers = workers;
        this.shifts = shifts;
        this.month = month;
        this.fullTimeHours = fullTimeHours;
        this.rules = rules;
    }
}

File: backend/src/main/java/com/graafik/model/ScheduleRequest.java
package com.graafik.model;

import java.util.List;

public class ScheduleRequest {
    private List<Worker> workers;
    private List<Shift> shifts;
    private int month;
    private long fullTimeMinutes;

    public ScheduleRequest() {}

    public List<Shift> getShifts() {return shifts;}
    public void setShifts(List<Shift> shifts) { this.shifts = shifts; }

    public List<Worker> getWorkers() { return workers; }
    public void setWorkers(List<Worker> workers) { this.workers = workers; }

    public int getMonth() { return month; }

    public long getFullTimeMinutes() { return fullTimeMinutes; }

}